content/blog/grpc-web-interceptor.md

```diff
@@ -52,88 +36,58 @@ const UnaryInterceptor = function() {};
  *     invoker
  * @return {!Promise<!UnaryResponse<RESPONSE>>}
  */
-
 UnaryInterceptor.prototype.intercept = function(request, invoker) {};
 ```
 
-Unary/StreamInterceptor is designed as an interface with one method to be
-implemented. A dedicated class for interceptors will have better scalability
-since we usually want to implement it with some complex functionality. This will
-also make things a lot easier for gRPC-Web to be embedded in other client
-platforms.
+The `intercept()` method takes two parameters:
 
-The intercept method takes in two parameters:
+- A `request` of type [grpc.web.Request][]
+- An `invoker`, which performs the actual RPC when invoked
 
-- **[grpc.web.Request](https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/request.js)**:
-  A wrapper class of RequestMessage, MethodDescriptor, Metadata, CallOptions.
-- **Invoker function**: This function will make the actual RPC call and return
-  the response.
+The `StreamInterceptor` interface declaration is similar, except that the
+`invoker` return type is `ClientReadablaStream` instead of `Promise`. For
+implementation details, see [interceptor.js][].
 
-Besides grpc.web.Request, we’ve introduced another new class here:
+## What can I do with an interceptor?
 
-- **[grpc.web.UnaryResponse](https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/unaryresponse.js)**:
-  A wrapper class of response message, status and response metadata, so that we
-  could handle the response metadata(headers) via interceptors.
+Basically, an interceptor allows you to do the following:
 
-## Configure interceptors
```

Comment:
This section is moved to after the examples so that the reader gets to see what an interceptor looks like before seeing code that makes use of interceptor instances.